frontend/catalyst/pennylane_extensions.py

```diff
@@ -2015,22 +2032,31 @@ def circuit():
     if len(wires) != 1:
         raise TypeError(f"One classical argument (a wire) is expected, got {wires}")
 
-    in_classical_tracers = wires
+    # Copy, so wires remain unmodified
+    in_classical_tracers = wires.copy()
 
-    # Check the postselect value. If given, add it to the classical tracers list
-    if postselect is not None:
-        if postselect not in [0, 1]:
-            raise TypeError(f"postselect must be '0' or '1', got {postselect}")
-        in_classical_tracers.append(postselect)
+    if postselect is not None and postselect not in [0, 1]:
+        raise TypeError(f"postselect must be '0' or '1', got {postselect}")
+    in_classical_tracers.append(postselect)
 
     # assert len(ctx.trace.frame.eqns) == 0, ctx.trace.frame.eqns
-    out_classical_tracer = new_inner_tracer(ctx.trace, get_aval(True))
+    m = new_inner_tracer(ctx.trace, get_aval(True))
     MidCircuitMeasure(
         in_classical_tracers=in_classical_tracers,
-        out_classical_tracers=[out_classical_tracer],
+        out_classical_tracers=[m],
         regions=[],
     )
-    return out_classical_tracer
+
+    # If reset was requested, reset qubit only if the measurement result was 1
+    if reset:
+
+        @cond(m)
+        def reset():
+            qml.PauliX(wires=wires)
+
+        reset()
```

Comment:
nice! what was the issue when we tried this patch yesterday then?

Reply:
`in_classical_tracers = wires` was a reference to `wires`. Then later we do `in_classical_tracers.append(postselect)`, which modifies `wires`! It didn't matter before because we weren't using wires after the measurement. But now it matters. So, now we do `in_classical_tracers = wires.copy()` to avoid the whole reference issue. Thanks @rmoyard for the hint!